slider/beatmap.py

```diff
@@ -660,6 +661,27 @@ def tick_points(self):
             ),
         )
 
+    @lazyval
+    def true_tick_points(self):
+        tick_points = self.tick_points
+        # curve() takes in a percentage of how far along we want the point.
+        # Take away the offset from the total length of the slider to get
+        # the percentage of the slider we want the point at.
+        true_end_time = (
+            self.end_time - self.LEGACY_LAST_TICK_OFFSET
+        )
+
+        duration = true_end_time - self.time
+        real_duration = self.end_time - self.time
+
+        ratio = duration / real_duration
+        curve_point = int(self.length * ratio)
+        pos = self.curve(curve_point / self.length)
```

Comment:
I don't understand this calculation. Shouldn't this be as simple as:

```python
duration = self.end_time - self.time
pos = self.curve((duration - self.LEGACY_LAST_TICK_OFFSET) / duration)
```

put another way: what is distinguishing between duration and real_duration achieving in this calculation?

Reply:
It may have been written in an odd way, the idea here was to seperate out and name the different variables.

Instead of using the temporary your code snippet produces, duration is accounting for the legacy last tick offset. I think renaming here will help.

Reply:
but is accounting for the legacy last tick offset the correct thing to do? the current code and the code I pasted above give different values. It's difficult to say at a glance which one [lazer uses](https://github.com/ppy/osu/blob/5a1940facf2649bc6b2892965c27b8c62a41988f/osu.Game/Rulesets/Objects/SliderEventGenerator.cs#L81) given the differences in abstractions, but I would like to see this confirmed one way or the other. Probably by double checking against lazer output on this map.